File: Memo01/app/src/main/java/com/callor/memo/dao/MemoDao.java
package com.callor.memo.dao;


import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.callor.memo.domain.MemoVO;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Flowable;

@Dao
public interface MemoDao {

    @Query("SELECT * FROM tbl_memo")
    Flowable<MemoVO> getAll();

     @Query("SELECT * FROM tbl_memo WHERE rowid IN (:rowid)")
     List<MemoVO> loadAllByIds(int[] rowid);

     @Query("SELECT * FROM  tbl_memo WHERE m_text LIKE :m_text")
     MemoVO findByText(String m_text);

     @Insert
     void insertAll(MemoVO... memoList);

     @Insert
     void insert(MemoVO memoVO);

     @Insert(onConflict = OnConflictStrategy.REPLACE)
     Completable insertMemo(MemoVO memoVO);

     @Delete
     void delete(long rowid);


}

File: MemoKt01/app/src/main/java/com/callor/memo/dao/MemoDao.kt
package com.callor.memo.dao


import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

import com.callor.memo.domain.MemoVO

import io.reactivex.Completable
import io.reactivex.Flowable

@Dao
interface MemoDao {

    @get:Query("SELECT * FROM tbl_memo")
    val all: Flowable<MemoVO>

    @Query("SELECT * FROM tbl_memo WHERE rowid IN (:rowid)")
    fun loadAllByIds(rowid: IntArray): List<MemoVO>

    @Query("SELECT * FROM  tbl_memo WHERE m_text LIKE :m_text")
    fun findByText(m_text: String): MemoVO

    @Insert
    fun insertAll(vararg memoList: MemoVO)

    @Insert
    fun insert(memoVO: MemoVO)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertMemo(memoVO: MemoVO): Completable

    @Delete
    fun delete(rowid: Long)


}
